core/src/main/java/jenkins/slaves/EncryptedSlaveAgentJnlpFile.java

```diff
@@ -70,7 +70,7 @@ public void generateResponse(StaplerRequest req, StaplerResponse res, Object nod
             final ByteArrayOutputStream baos = new ByteArrayOutputStream();
             StaplerResponse temp = new ResponseImpl(req.getStapler(), new HttpServletResponseWrapper(res) {
                 @Override public ServletOutputStream getOutputStream() throws IOException {
-                    return new FilterServletOutputStream(baos);
+                    return new FilterServletOutputStream(baos, super.getOutputStream());
```

Comment:
Could it cause issues for downstream deps?

Reply:
I guess? Ask @jtnord who requested this.

Reply:
See discussion: https://github.com/stapler/stapler/pull/131#discussion_r140082051

Reply:
this is the wrong change IIUC.  that wraps the  output stream of the response - which will return false information.

The correct fix is https://github.com/jenkinsci/jenkins/pull/3033/files#diff-8991f368b3f7346bf83c7081c719c809R118

Reply:
FilterServletOutputStream is likely to be removed in a future version - its only there to do home grown GZIP which is causing more harm that good, and other places have abused it to do things like this...